src/Parser/Core/Modules/StatTracker.js

```diff
@@ -0,0 +1,335 @@
+    };
+    this._stats = this._startingStats;
+
+    debug && this._debugPrintStats(this._stats);
+  }
+
+  /*
+   * Stat rating at pull.
+   * Should be identical to what you get from Combatant.
+   */
+  get startingStrengthRating() {
+    return this._startingStats.str;
+  }
+  get startingAgilityRating() {
+    return this._startingStats.agi;
+  }
+  get startingIntellectRating() {
+    return this._startingStats.int;
+  }
+  get startingStaminaRating() {
+    return this._startingStats.stam;
+  }
+  get startingCritRating() {
+    return this._startingStats.crit;
+  }
+  get startingHasteRating() {
+    return this._startingStats.haste;
+  }
+  get startingMasteryRating() {
+    return this._startingStats.mastery;
+  }
+  get startingVersRating() {
+    return this._startingStats.vers;
+  }
+  get startingAvoidanceRating() {
+    return this._startingStats.avoidance;
+  }
+  get startingLeechRating() {
+    return this._startingStats.leech;
+  }
+  get startingSpeedRating() {
+    return this._startingStats.speed;
+  }
+
+  /*
+   * Current stat rating, as tracked by this module.
+   */
+  get currentStrengthRating() {
+    return this._stats.str;
+  }
+  get currentAgilityRating() {
+    return this._stats.agi;
+  }
+  get currentIntellectRating() {
+    return this._stats.int;
+  }
+  get currentStaminaRating() {
+    return this._stats.stam;
+  }
+  get currentCritRating() {
+    return this._stats.crit;
+  }
+  get currentHasteRating() {
+    return this._stats.haste;
+  }
+  get currentMasteryRating() {
+    return this._stats.mastery;
+  }
+  get currentVersRating() {
+    return this._stats.vers;
+  }
+  get currentAvoidanceRating() {
+    return this._stats.avoidance;
+  }
+  get currentLeechRating() {
+    return this._stats.leech;
+  }
+  get currentSpeedRating() {
+    return this._stats.speed;
+  }
+
+  /*
+   * For percentage stats, the percentage you'd have with zero rating.
+   * These values don't change.
+   */
+  get baseCritPercentage() {
+    return 0.08; // TODO is this the same for all classes?
+  }
+  get baseHastePercentage() {
+    return 0;
+  }
+  get baseMasteryPercentage() {
+    switch (this.combatants.selected.spec) {
+      case SPECS.HOLY_PALADIN:
+        return 0.12;
+      case SPECS.HOLY_PRIEST:
+        return 0.05;
+      case SPECS.RESTORATION_SHAMAN:
+        return 0.24;
+      case SPECS.ENHANCEMENT_SHAMAN:
+        return 0.2;
+	    case SPECS.RESTORATION_DRUID:
+	      return 0.048;
+      default:
+        throw new Error('Mastery hasn\'t been implemented for this spec yet.');
+    }
+  }
+  get baseVersPercentage() {
+    return 0;
+  }
+  get baseAvoidancePercentage() {
+    return 0;
+  }
+  get baseLeechPercentage() {
+    return 0;
+  }
+  get baseSpeedPercentage() {
+    return 0;
+  }
+
+  /*
+   * For percentage stats, this is the multiplier to go from rating to percent (expressed from 0 to 1)
+   * These values don't change.
+   */
+  get critRatingToPercent() {
```

Comment:
I think this should be `critRatingPerPercent` so that it can be used for more things that just calculating the percentage.

Reply:
It's a conversion factor, and 'perPercent' sounds stupid! Seriously though, I originally had it like you suggest and found myself just dividing everywhere. If you really need the perPercent, you can just divide by it instead of multiply.

Reply:
I think it makes more sense to do 12,000 / 400 to get the crit percentage than 12,000 * number I can't remember nor type. Isn't this most of the applications of this value?